src/rust/plugin-registry/src/server.rs

```diff
@@ -125,16 +187,33 @@ impl PluginRegistryService for PluginRegistry {
         todo!()
     }
 
+    #[tracing::instrument(skip(self, request), err)]
     async fn get_generators_for_event_source(
         &self,
-        _request: Request<GetGeneratorsForEventSourceRequestProto>,
+        request: Request<GetGeneratorsForEventSourceRequestProto>,
     ) -> Result<Response<GetGeneratorsForEventSourceResponseProto>, Status> {
-        // Stub implementation, for integration test smoke-test purposes.
-        // Replace with a real implementation soon!
-        let message = _request.into_inner();
-        Ok(Response::new(GetGeneratorsForEventSourceResponseProto {
-            plugin_ids: [message.event_source_id].into_iter().flatten().collect(),
-        }))
+        let request = request.into_inner();
+        let request =
+            GetGeneratorsForEventSourceRequest::try_from(request).expect("Invalid message");
+
+        match self.get_generators_for_event_source(request).await {
+            Ok(response) => {
+                tracing::debug!(
+                    message="Successfully retrieved generators for event source",
+                    plugin_ids=?response.plugin_ids,
```

Comment:
Would it be beneficial to attach anything else from the response, or maybe just the whole response itself?